Pegasus/include/Geometry.hpp

```diff
@@ -677,31 +695,43 @@ inline void Initialize<Box, Box>(SimpleShape const* a, SimpleShape const* b, Cac
     auto bBox = static_cast<Box const *>(b);
     auto cache = static_cast<Cache<Box, Box>*>(cacheBase);
 
-    cache->aMassCenter = aBox->getCenterOfMass();
+    cache->aMassCenter = aBox->GetCenterOfMass();
     aBox->GetAxes(cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2]);
     cache->aBoxAxes = {cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2],
         cache->aBoxAxes[0] * -1.0, cache->aBoxAxes[1] * -1.0, cache->aBoxAxes[2] * -1.0};
     CalculateBoxVertices(cache->aBoxAxes[0], cache->aBoxAxes[1], cache->aBoxAxes[2], cache->aBoxVertices.begin());
-    std::for_each(cache->aBoxVertices.begin(), cache->aBoxVertices.end(), [cache](auto& v) { v += cache->aMassCenter; });
+    std::for_each(cache->aBoxVertices.begin(), cache->aBoxVertices.end(), [cache](auto& v)
```

Comment:
Nope, accumulate will calculate the sum of the vertices, but this code performs transformation by translation.